opensquirrel/merger/general_merger.py

```diff
@@ -79,6 +86,41 @@ def try_name_anonymous_bloch(bsr: BlochSphereRotation) -> BlochSphereRotation:
     return bsr
 
 
+def _flatten_list(list_to_flatten: list[list[Any]]) -> list[Any]:
+    return functools.reduce(operator.iadd, list_to_flatten, [])
+
+
+def merge_barriers(statement_list: list[Statement]) -> list[Statement]:
+    """Receives a list of statements.
+    Returns an ordered version of the input list of statements where groups of barriers are merged together,
+    and placed as late in the list as possible.
+
+    Args:
+        statement_list: list of statements
+
+    Returns:
+        Statement list with the barriers merged.
+    """
+    barrier_list: list[Barrier] = []
+    ordered_statement_list: list[Statement] = []
+    for _, statement in enumerate(statement_list):
+        if isinstance(statement, Barrier):
+            barrier_list.append(statement)
+        else:
+            if len(barrier_list) > 0 and hasattr(statement, "get_qubit_operands"):
```

Comment:
```suggestion
        if isinstance(statement, Comment):
            continue
        if isinstance(statement, Barrier):
            barrier_list.append(statement)
        else:
            if len(barrier_list) > 0:
                assert(hasattr(statement, "get_qubit_operands"))
```

This is only needed because, at the moment, we are treating comments as statements, right?
In that case, I would just continue looping when we find a comment, and remove the check.
I have opened a Backlog task for changing Comment's to not being Statement's.

Reply:
It still gives mypy errors because it does not believe you handled all cases. I will revert and change this once we allow the `Statement` class to have `get_qubit_operands` as an abstract method